dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/WebRumEventContextProvider.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+
+internal class WebRumEventContextProvider {
+
+    fun getRumContext(): RumContext? {
+        val rumContext = GlobalRum.getRumContext()
+        return if (rumContext.applicationId == RumContext.NULL_UUID ||
+            rumContext.sessionId == RumContext.NULL_UUID
+        ) {
+            devLogger.w(RUM_NOT_INITIALIZED_WARNING_MESSAGE)
+            sdkLogger.e(RUM_NOT_INITIALIZED_ERROR_MESSAGE)
```

Comment:
You can still spam this calling multiple time the `getRumContext()` method.

Reply:
yes but this will be called only from the `WebRumEventConsumer` and there we have the `rumFeatureDisabled` flag which protects it.

Reply:
Should we move the flag inside the `WebRumContextProvider` instead ?

Reply:
I think I made a comment earlier in that sense yes.

Reply:
Yup, it is done